### PR TITLE
Implement restful for InstallLogsSupport

### PR DESCRIPTION
**Pull request by ping035627**
The PR implement restful for InstallLogsSupport in apiserver.go.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/29981)

### CODE REVIEW

**Comment by nikhiljindal** [comment]
@k8s-bot ok to test

**Comment by ping035627** [comment]
@nikhiljindal the pr has passed the test, need lgtm, thanks.

File: api/swagger-spec/resourceListing.json [hunk 1]
```diff
@@ -2,6 +2,10 @@
   "swaggerVersion": "1.2",
   "apis": [
    {
+    "path": "/logs",
```
Comment (nikhiljindal) [inline api/swagger-spec/resourceListing.json]: There should be a new logs.json file. You will need to update update-swagger-spec.sh to include it.

**Comment by nikhiljindal** [comment]
This promotes '/logs' endpoint as we will now start surfacing it in the discovery API.
Should be fine but will be good to get confirmation from @lavalamp .
Also cc @kubernetes/sig-api-machinery .

**Comment by nikhiljindal** [comment]
Also, can you please add a test to https://github.com/kubernetes/kubernetes/blob/master/pkg/master/master_test.go to make sure that /logs is accessible on the server and the response is as expected?

**Comment by smarterclayton** [comment]
ok to test

File: api/swagger-spec/resourceListing.json (hunk 1 above)
Comment (ping035627) [inline api/swagger-spec/resourceListing.json]: ok,  I have modified.

**Comment by ping035627** [comment]
I add  a test to apiserver_test.go,  I think it's perhaps more suitable.  Can you review again? thanks.

File: api/swagger-spec/logs.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "swaggerVersion": "1.2",
+  "apiVersion": "",
+  "basePath": "https://10.10.10.10:6443",
+  "resourcePath": "/logs",
+  "apis": [
+   {
+    "path": "/logs/{logpath}",
+    "description": "get logs files",
```
Comment (nikhiljindal) [inline api/swagger-spec/logs.json]: get logs files -> get log files?

**Comment by nikhiljindal** [comment]
LGTM, modulo a minor comment

**Comment by ping035627** [comment]
I have modified the comment, but the "Jenkins GCE Node e2e" failed. It shouldn't because of my work. Can test again? thanks.

**Comment by nikhiljindal** [comment]
@k8s-bot unit test this issue: #IGNORE
@k8s-bot gce e2e node test this issue: #IGNORE

**Comment by ping035627** [comment]
It seems not trigger another test?

**Comment by ping035627** [comment]
@nikhiljindal  It  seems  not  trigger another  test run,  can  test again?

**Comment by ping035627** [comment]
@nikhiljindal can you help to trigger another test? the Jenkins GCE Node e2e failed, but seems not because of my work. thanks.

**Comment by nikhiljindal** [comment]
Sorry @ping035627 . I issued the wrong command.
@k8s-bot gce node e2e test this issue: #IGNORE

(found via https://github.com/kubernetes/test-infra/issues/269#issuecomment-231362400)

**Comment by ping035627** [comment]
yes, I see, but it doesn't trigger a new test, I will execute "rebase" to trigger a test, :)

**Pull request merged**